org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/impl/XbaseInterpreter.java

```diff
@@ -114,8 +114,16 @@
  */
 public class XbaseInterpreter implements IExpressionInterpreter {
 	
+	/**
+	 * Encapsulates the return value of an {@link XReturnExpression}
```

Comment:
```suggestion
Encapsulates the return value of an {@link XReturnExpression}.

To be able to break up the control flow of the interpreter, this class inherits from {@link RuntimeException}.
Clients that catch the ReturnValue, should usually re-throw it further. See also {@link _doEvaluate(XTryCatchFinallyExpression, ..)} where the finally clause is evaluated after the return value has been computed but the original return value is propagated further, if the finally block completes normally.
```

I would be fine with keeping it concise and simply remove the part about the TryCatchFinallyExpression.

Reply:
Thanks for the explanation @szarnekow ! I updated the PR accordingly.